airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/discover/DiscoverIntegrationTest.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.discover
+
+import io.airbyte.cdk.command.ConfigurationSpecification
+import io.airbyte.cdk.command.FeatureFlag
+import io.airbyte.cdk.load.command.Property
+import io.airbyte.cdk.load.test.util.ConfigurationUpdater
+import io.airbyte.cdk.load.test.util.FakeConfigurationUpdater
+import io.airbyte.cdk.load.test.util.FakeDataDumper
+import io.airbyte.cdk.load.test.util.IntegrationTest
+import io.airbyte.cdk.load.test.util.NoopDestinationCleaner
+import io.airbyte.cdk.load.test.util.NoopExpectedRecordMapper
+import io.airbyte.cdk.load.test.util.destination_process.NonDockerizedDestinationFactory
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import java.util.regex.Pattern
+import kotlin.test.assertEquals
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertAll
+
+data class DiscoverTestConfig(
+    val configContents: String,
+    val featureFlags: Set<FeatureFlag> = emptySet(),
+    val name: String? = null,
+)
+
+abstract class DiscoverIntegrationTest<T : ConfigurationSpecification>(
+    val successConfigFilenames: List<DiscoverTestConfig>,
+    val failConfigFilenamesAndFailureReasons: Map<DiscoverTestConfig, Pattern>,
+    additionalMicronautEnvs: List<String> = emptyList(),
+    micronautProperties: Map<Property, String> = emptyMap(),
+    configUpdater: ConfigurationUpdater = FakeConfigurationUpdater,
+) :
+    IntegrationTest(
+        additionalMicronautEnvs = additionalMicronautEnvs,
+        dataDumper = FakeDataDumper,
+        destinationCleaner = NoopDestinationCleaner,
+        recordMangler = NoopExpectedRecordMapper,
+        configUpdater = configUpdater,
+        micronautProperties = micronautProperties,
+    ) {
+    @BeforeEach
+    fun setupProcessFactory() {
+        if (destinationProcessFactory is NonDockerizedDestinationFactory) {
+            destinationProcessFactory.injectInputStream = false
+        }
+    }
+
+    @Test
+    open fun testSuccessConfigs() {
+        for (tc in successConfigFilenames) {
+            val updatedConfig = updateConfig(tc.configContents)
+            val process =
+                destinationProcessFactory.createDestinationProcess(
+                    "discover",
+                    configContents = updatedConfig,
+                    featureFlags = tc.featureFlags.toTypedArray(),
+                    micronautProperties = micronautProperties,
+                )
+            runBlocking { process.run() }
+            val messages = process.readMessages()
+            val checkMessages = messages.filter { it.type == AirbyteMessage.Type.CONNECTION_STATUS }
```

Comment:
I'm guessing this is just a placeholder, and eventually we'll be looking for a catalog message?

Reply:
Good catch, I updated the test to be more relevant.
I am pretty we might adjust this as we go and add tests.